### PR TITLE
Fetching users songs only instead of all. Removing songs from state

### PR DESCRIPTION
**Pull request by escobarjonatan**
Removes songs from app state as they are not needed at that level
Displays songs in library page owned by current user

### CODE REVIEW

File: src/modules/session/thunks.ts [hunk 1]
```diff
@@ -52,12 +51,6 @@ export const updateInitialProfile = createAsyncThunk(
 export const getInitialData = createAsyncThunk(
   "session/getInitialData",
   async (_, { dispatch }) => {
-    const songsResponse = await dispatch(songApi.endpoints.getSongs.initiate());
-
-    if ("error" in songsResponse) {
-      return;
-    }
-
```
Comment (escobarjonatan) [inline src/modules/session/thunks.ts:60]: Removing this as there is no need to request new songs every time we get initial data. Songs should only be requested when the user is on the library page.

Comment (escobarjonatan) [reply src/modules/session/thunks.ts:60]: getInitialData gets called on every single page load and we dont need to do that. We can keep it to fetch only on the library where we actually use the data

**Review by dmkirshon — APPROVED** [review]
LGTM, just as two side comments:

1. I know you didn't touch it this branch but looks like the `NoSongsYet` Component needs to be updated to match mocks in FIgma, task needs to be added either to QA for MVP task or Library 1.3 overall task
2. Did you test uploading your own song to get the Library to display correctly with this new change? 
Not sure if still CORS issues but when attempting to upload a new song via local and the preview url I received the following toast: `An error while uploading your song` and Console error: `Failed to load resource: the server responded with a status of 404 ()`
Used an 30 second  .wav file and a mockup jpg, both showed success green check marks. Do I need to start the Mint Song process because it's waiting for the agreement?

**Comment by escobarjonatan** [comment]
> LGTM, just as two side comments:
> 
> 1. I know you didn't touch it this branch but looks like the `NoSongsYet` Component needs to be updated to match mocks in FIgma, task needs to be added either to QA for MVP task or Library 1.3 overall task
> 2. Did you test uploading your own song to get the Library to display correctly with this new change?
>    Not sure if still CORS issues but when attempting to upload a new song via local and the preview url I received the following toast: `An error while uploading your song` and Console error: `Failed to load resource: the server responded with a status of 404 ()`
>    Used an 30 second  .wav file and a mockup jpg, both showed success green check marks. Do I need to start the Mint Song process because it's waiting for the agreement?

@dmkirshon 
1. Good idea, adding it as a subtask under Library 1.3: https://app.clickup.com/t/863gakfcx
2. I tried the upload flow a few weeks/month back and it was working. It looks like it's not now. I will look into this as part of this PR. 
It looks like we are making 2 of the same upload calls, the first one we get a 200 and the second one we get a 404. So if you go to your library after you get the error, you will see the "failed" song there 🪄

**Comment by escobarjonatan** [comment]
@dmkirshon @scandycuz Turns out the endpoint for getAudioUploadUrl changed from `/v1/songs/${songId}/upload` to `/v1/songs/${songId}/audio` changing that fixed the upload issue. However, we havent handled what happens after a successful upload so it will stay on the same page with the fields filled out

**Comment by dmkirshon** [comment]
> @dmkirshon @scandycuz Turns out the endpoint for getAudioUploadUrl changed from `/v1/songs/${songId}/upload` to `/v1/songs/${songId}/audio` changing that fixed the upload issue. However, we havent handled what happens after a successful upload so it will stay on the same page with the fields filled out

All of this is still probably related to the progress circles still spinning, aka no stream URL provided, on the test audio files just uploaded, or is there a transcoder backend issue that needs to be addressed?

**Comment by escobarjonatan** [comment]
> > @dmkirshon @scandycuz Turns out the endpoint for getAudioUploadUrl changed from `/v1/songs/${songId}/upload` to `/v1/songs/${songId}/audio` changing that fixed the upload issue. However, we havent handled what happens after a successful upload so it will stay on the same page with the fields filled out
> 
> All of this is still probably related to the progress circles still spinning, aka no stream URL provided, on the test audio files just uploaded, or is there a transcoder backend issue that needs to be addressed?

right, there's currently a thread about this in the [#devops](https://projectnewmgroup.slack.com/archives/C04P2C60JS1/p1679724329173469) channel in slack. Will merge this PR since topic is not related to this PR. If there is a change we need to do on our end, it can be its own PR

**Pull request merged**